runner/runner.go

```diff
@@ -231,60 +264,77 @@ func (r *Runner) run(ctx context.Context, checkMap map[string]*Check, list bool,
 		}
 	}
 
+	responseChannel := make(chan *Response, len(currentCheckList))
+	wg := &sync.WaitGroup{}
+	
 	// main loop to get the checks info.
 	for _, name := range currentCheckList {
-		resp := &Response{
-			name: name,
-		}
+		wg.Add(1)
+		xname := name
+		go func(xname string, responseChannel chan *Response) {
+			defer wg.Done()
+			currentCheck, ok := checkMap[xname]
+			if !ok {
+				combinedResponse.errs["check not found"] = &Response{
```

Comment:
Pretty sure this is a race condition. You are assigning to `errs` from multiple goroutines. The coolest way to fix this IMO is to add a method on combinedResponse called `SetErr()` or similar and have that method take a mutex lock when it enters and defers the unlock.